osu.Game/Rulesets/Objects/Types/IHasSliderVelocity.cs

```diff
@@ -16,5 +18,30 @@ public interface IHasSliderVelocity
         double SliderVelocityMultiplier { get; set; }
 
         BindableNumber<double> SliderVelocityMultiplierBindable { get; }
+
+        /// <summary>
+        /// Introduces floating-point errors to post-multiplied beat length for rulesets that depend on it.
+        /// </summary>
+        public double GetPrecisionAdjustedBeatLength(TimingControlPoint timingControlPoint, string rulesetShortName)
+        {
+            double sliderVelocityAsBeatLength = -100 / SliderVelocityMultiplier;
+
+            // Note: In stable, the division occurs on floats, but with compiler optimisations turned on actually seems to occur on doubles via some .NET black magic (possibly inlining?).
+            double bpmMultiplier;
+
+            switch (rulesetShortName)
+            {
+                case "taiko":
+                case "mania":
+                    bpmMultiplier = sliderVelocityAsBeatLength < 0 ? Math.Clamp((float)-sliderVelocityAsBeatLength, 10, 10000) / 100.0 : 1;
+                    break;
+
+                default:
+                    bpmMultiplier = sliderVelocityAsBeatLength < 0 ? Math.Clamp((float)-sliderVelocityAsBeatLength, 10, 1000) / 100.0 : 1;
+                    break;
+            }
```

Comment:
If this is intended as a method for all rulesets to use, do you think we instead specify explicit osu and catch branches and leave the `default` branch as simple/lossless as possible?

Reply:
Dunno. It may be sunk cost fallacy from all the overnight sheet running but I tend to think that this is okay. The fact that we're using legacy `.osu` format and encoding slider velocity into files in one of the most opaque ways I can imagine also contributes to this feeling - like if we were redoing this to be simpler, then we'd redefine the actual way the thing is _stored_ to not require the weird division by 100.

Can wait until further work in my opinion.

Reply:
I mean it from a different angle - this method is essentially a legacy method that pollutes an interface intended for any ruleset to use.

I can imagine any ruleset consumer would use this as a shortcut for `timingPoint.BeatLength * SliderVelocityMultiplier` without understanding the consequences of this - the precision and the weird clamping behaviour.

How do you explain the existence and purpose of this method to a ruleset consumer? I believe it should be more properly indicated.

Reply:
I see.

For one, it has to be public API, so that's that, but what'd work for you? More xmldoc? Moving to extension method? Marking as deprecated with inline disables?

Reply:
I've moved it to an extension method. I've added more xmldoc, and I've added a `throw` on non-legacy use.